packages/fx-core/src/component/driver/deploy/azure/impl/azureDeployImpl.ts

```diff
@@ -170,6 +170,34 @@ export abstract class AzureDeployImpl extends BaseDeployImpl {
     azureResource: AzureResourceInfo,
     azureCredential: TokenCredential
   ): Promise<AzureUploadConfig> {
+    try {
+      const defaultScope = "https://management.azure.com/.default";
+      const token = await azureCredential.getToken(defaultScope);
+      if (token) {
+        this.logger.info("Get AAD token successfully. Upload zip package through AAD Auth mode.");
+        return {
+          headers: {
+            "Content-Type": "application/octet-stream",
+            "Cache-Control": "no-cache",
+            Authorization: `Bearer ${token.token}`,
+          },
+          maxContentLength: Infinity,
+          maxBodyLength: Infinity,
+          timeout: DeployConstant.DEPLOY_TIMEOUT_IN_MS,
+        };
+      } else {
+        this.logger.info(
+          "Get AAD token failed. AAD Token is empty. Upload zip package through basic auth mode. Please check your Azure credential."
+        );
+      }
+    } catch (e) {
+      this.logger.info(
+        `Get AAD token failed with error: ${JSON.stringify(
+          e
+        )}. Upload zip package through basic auth mode.`
+      );
+    }
+
```

Comment:
Should we update the bicep file to disable basic authentication? If we make this change, the basic authentication logic that follows will no longer work.

PS: When we create an App Service with Basic Authentication enabled, it may trigger an alert.

Reply:
I will remove that code snippet after the ARM templates are changed.